### PR TITLE
feat(cli): Add insecure option for https connections

### PR DESCRIPTION
**Pull request by elizabethhealy**
Currently --insecure forces http plaintext. Instead have --insecure use an ssl factory that does not check certs. Also add a flag for http plaintext (-h).

### CODE REVIEW

**Comment by elizabethhealy** [comment]
xtest failing due to flag changes but i have a PR to fix it after this merges https://github.com/opentdf/tests/pull/185

File: cmdline/src/main/java/io/opentdf/platform/Command.java [hunk 1]
```diff
@@ -84,10 +93,17 @@ void encrypt(
...
+            SSLFactory sslFactory = SSLFactory.builder()
+            .withUnsafeTrustMaterial() // Trust all certificates
+            .build();
+            builder.sslFactory(sslFactory);
+        }
+        
+        return  builder.platformEndpoint(platformEndpoint)
                 .clientSecret(clientId, clientSecret)
-                .useInsecurePlaintextConnection(insecure)
+                .useInsecurePlaintextConnection(plaintext)
```
Comment (strantalis) [inline cmdline/src/main/java/io/opentdf/platform/Command.java:106]: Will this ever conflict with the sslFactory?

Comment (elizabethhealy) [reply cmdline/src/main/java/io/opentdf/platform/Command.java:106]: sslfactory is only used if plaintext isnt set, when plaintext set it just defaults to http and doesnt use the factory
https://github.com/opentdf/java-sdk/blob/7069b1ba5ddc7098a7997cdcfcfdaddb6f899b63/sdk/src/main/java/io/opentdf/platform/sdk/SDKBuilder.java#L227-L237

Comment (elizabethhealy) [reply cmdline/src/main/java/io/opentdf/platform/Command.java:106]: if neither are set it looks like it just does `channelBuilder = ManagedChannelBuilder.forTarget(endpoint);`

**Review by strantalis — APPROVED** [review]

**Pull request merged**